src/rt/osx_tls.c

```diff
@@ -0,0 +1,49 @@
+/**
+ * Helpers for determining TLS memory ranges on OS X.
+ *
+ * This unfortunately cannot be entirely done in D, as the OS X API uses
+ * the Apple-specific blocks C extension.
+ *
+ * Copyright: David Nadlinger, 2012.
+ * License:   <a href="http://www.boost.org/LICENSE_1_0.txt">Boost License 1.0</a>.
+ * Authors:   David Nadlinger
+ */
+
+#ifndef __BLOCKS__
```

Comment:
There's a theoretical possibility that someone will try to compile this with GCC. So perhaps check for both OS X and blocks? @klickverbot will it be possible for LDC to use this functionality? In that case what, preprocessor constant can I use to cover all Apple platforms?